sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/DocumentBuffer.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using Azure.Monitor.OpenTelemetry.LiveMetrics.Internals.Diagnostics;
+using Azure.Monitor.OpenTelemetry.LiveMetrics.Models;
+
+namespace Azure.Monitor.OpenTelemetry.LiveMetrics
+{
+    /// <summary>
+    /// Manages a thread-safe collection of DocumentIngress objects with a fixed capacity.
+    /// </summary>
+    internal class DocumentBuffer
+    {
+        private readonly ConcurrentQueue<DocumentIngress> _documents = new();
+        private readonly int _capacity = 20;
+        // ConcurrentQueue<T>.Count is not used because it is not an O(1) operation. Instead, we use a separate counter.
+        // Atomic counter for the number of documents in the queue.
+        private int _count = 0;
+
+        public void Add(DocumentIngress document)
+        {
+            // Ensure the queue does not exceed capacity.
+            if (Interlocked.CompareExchange(ref _count, 0, 0) < _capacity)
+            {
+                _documents.Enqueue(document);
+                Interlocked.Increment(ref _count);
+            }
+            else
+            {
+                // TODO: Log dropped documents message to event source.
```

Comment:
Is this acceptable?

This means that the metric values user would see could be higher than the actual documents they see on the side panel if the # of items exceed `20`.  

Also, `20` might a low number considering we would be adding the logs to the same buffer as well.